pallets/ocex/src/lib.rs

```diff
@@ -380,10 +383,12 @@ pub mod pallet {
 				signature.verify(bytes.as_slice(), &enclave),
 				Error::<T>::EnclaveSignatureVerificationFailed
 			);
-			<Withdrawals<T>>::insert(snapshot.snapshot_number, snapshot.withdrawals);
+			let mut withdrawal_map: BoundedBTreeMap<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit> = BoundedBTreeMap::try_from(snapshot.withdrawals.clone()).expect("Bounded BTreeMap could not be created");
```

Comment:
Are you sure that we need to panic here and not returning error?

Reply:
@felixfaisal there should be no possibility for a panic, even by a unhandled arithmetic operation is allowed in runtime code.

Reply:
Error Handling done ✅